### ISSUE TITLE
Questions about the warping part of the code

### ISSUE THREAD

**Issue by JerryPW** [HKUST-3DV/DIM-SLAM#12]
Hi ! 
Recently, I've been using these codes to complement my warping constrains. However, I've found something strange.
```
batch_patch_gt_color = (
    torch.nn.functional.grid_sample(
        batch_patch_gt_color.permute(0, 3, 1, 2),
        batch_patch_uv.view(batch_patch_uv.shape[0], batch_patch_gt_color.shape[0], -1, 2).permute(1, 0, 2, 3),
        mode="bilinear",
        padding_mode='border'
    )
    .permute(2, 0, 3, 1)
    .contiguous()
    .view(-1, 3)
)
```

Specifically, as I focus on 1 patch of image and tuning the patch_size into the whole picture, the output, i.e. batch_patch_gt_color, suppose to be exactly the same with input image. Whereas, the output is the input with rotate 90 degree and flipped. Here's the input and the output image:
<img width="428" alt="截屏2023-08-15 14 27 49" src="https://github.com/HKUST-3DV/DIM-SLAM/assets/103237744/466ead95-05a9-408a-b0f3-c9379f77ec8e">
<img width="430" alt="截屏2023-08-15 14 28 05" src="https://github.com/HKUST-3DV/DIM-SLAM/assets/103237744/cf7c1597-ee9e-4fa2-b281-2e6f4f05363e">

Do you know why is that happened？
Hoping for your reply soon!

**Comment by poptree** [comment]
Hi,

It seems like the `batch_patch_uv` is wrong. batch_patch_uv[...,0] should be the x-axis of the image, which is usually along the width.

Please refer to line 113 of `utils.py` , where the first entry is width.

BTW, LINE 120 has a similar coordinate error on width and height, so I rewrite a new function `get_samples_by_indices_batch` instead of using `get_samples_batch`.

**Comment by poptree** [comment]
You could find more detail on [pytorch grid sample](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html).

Hope it will be helpful.

**Comment by poptree** [comment]
Besides, you know, some individuals may feel uncomfortable when viewing medical images. While I believe the pictures you posted are synthetic, this won't alleviate the discomfort.

🙏Maybe you could replace it with other images next time. I will close this issue since I cannot stay on it anymore.

**Comment by JerryPW** [comment]
> Hi,
> 
> It seems like the `batch_patch_uv` is wrong. batch_patch_uv[...,0] should be the x-axis of the image, which is usually along the width.
> 
> Please refer to line 113 of `utils.py` , where the first entry is width.
> 
> BTW, LINE 120 has a similar coordinate error on width and height, so I rewrite a new function `get_samples_by_indices_batch` instead of using `get_samples_batch`.

Thanks a lot !!!!!!
I've been working on this problem for all day!!!
And I will notice the 'medical image problem' you just mentioned next time! Thanks for remind me that!
